### PR TITLE
feat(mariadb): drop 10.3 and add 11.0 support

### PR DESCRIPTION
**Pull request by WikiRik**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

MariaDB 10.3 is EOL since May 25th; https://mariadb.org/about/#maintenance-policy
The reason why the lowest version is MariaDB 10.4.6 is because that is the first stable release of 10.4 and the first that has the mariadb commands symlinked

### CODE REVIEW

File: packages/core/test/integration/json.test.ts [hunk 1]
```diff
@@ -213,13 +213,21 @@ describe('JSON Querying', () => {
...
-      // @ts-expect-error -- getDataValue does not support custom attributes
-      expect(orders[0].user.getDataValue('name')).to.equal(expectedResult);
+      if (dialectName === 'mariadb' && semver.lt(sequelize.getDatabaseVersion(), '10.5.2')) {
+        if (semver.gt(sequelize.getDatabaseVersion(), '10.4.21')) {
+          // @ts-expect-error -- getDataValue does not support custom attributes
+          expect(orders[0].user.getDataValue('name')).to.equal('"swen"');
+        }
+
+        // @ts-expect-error -- getDataValue does not support custom attributes
+        expect(orders[0].user.getDataValue('name')).to.deep.equal(Buffer.from('"swen"'));
```
Comment (WikiRik) [inline packages/core/test/integration/json.test.ts:226]: I don't really know where this comes from, but I'm not sure if we want to fix it or increase the minimum patch version to 10.4.22. @ephys any preference?

Release notes of 10.4.22 are here; https://mariadb.com/kb/en/mariadb-10422-release-notes/ with a link to the changelog

File: dev/mariadb/oldest/docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 services:
   mariadb-oldest:
     container_name: sequelize-mariadb-oldest
-    image: mariadb:10.3.34
+    image: mariadb:10.4.6
```
Comment (ephys) [inline dev/mariadb/oldest/docker-compose.yml:4]: There is no good reason to support an old patch version, this should be 10.4.30

File: packages/core/test/integration/json.test.ts (hunk 1 above)
Comment (ephys) [reply packages/core/test/integration/json.test.ts:226]: These changes are very weird, so I'll wait to see if changing the version to 10.4.30 helps with this

**Review by ephys — CHANGES_REQUESTED** [review]

File: dev/mariadb/oldest/docker-compose.yml (hunk 2 above)
Comment (WikiRik) [reply dev/mariadb/oldest/docker-compose.yml:4]: Sure, I'll update that and make the changes to the test. The buffer is not needed anymore on recent patch versions

Comment (WikiRik) [reply dev/mariadb/oldest/docker-compose.yml]: Done

**Comment by WikiRik** [comment]
It seems that the following test is not flaky, but just broken for MariaDB 11.0;
```
  1) [MARIADB] Model
       findAll
         groupedLimit
           on: belongsToMany
             [Flaky] works with computed order:

      AssertionError: expected [ 1, 3, 5, 4, 7 ] to deeply equal [ 1, 3, 5, 4 ]
      + expected - actual

         1
         3
         5
         4
      -  7
       ]
```

**Comment by ephys** [comment]
Let's update the connector and see if that has a positive impact on the test. Unlikely but who knows

**Comment by WikiRik** [comment]
The connector upgrade seems to have done the trick, not sure why though. We'll have to keep looking out for this test if it fails again in the future

**Review by ephys — APPROVED** [review]

**Comment by ephys** [comment]
I'll let you merge for the breaking change documentation

**Pull request merged**